### PR TITLE
Style Spec for sky and fog

### PR DESCRIPTION
**Pull request by acalcutt**
This is a attempt at bringing in the style spec code from https://github.com/maplibre/maplibre-gl-js/pull/1713 into this new style-spec package. This is needed to implement sky and fog in maplibre-gl-js. Most of this code what pulled from the previous pr by @prozessor13 , but I've made a few tweaks where I saw it was needed.

I am mainly submitting this because I would like to see sky implemented and this seems like a good first step. This goes with https://github.com/maplibre/maplibre-gl-js/pull/2974 , which I think still needs some work.

### CODE REVIEW

**Comment by HarelM** [comment]
Can you please link to this PR in the following issue, and preferably comment on the suggestion there:
https://github.com/maplibre/maplibre-style-spec/issues/163

**Comment by acalcutt** [comment]
Thinking about how best to test this. Do we have the ability to create a pre-release style spec package? There seems to be a lot of references to this package in maplibre-gl-js, I'm wondering if maybe making a pre-release with these changes would be easier than making my own package like I am trying now.

**Comment by HarelM** [comment]
@drwestco I think you were are to do that, can you share your experience in a manual or a readme of some sort?

File: src/reference/v8.json [hunk 1]
```diff
@@ -63,6 +63,13 @@
         "intensity": 0.4
       }
     },
+    "sky": {
+      "type": "sky",
+      "doc": "The map sky.",
+      "example": {
+          "sky-color": "#199EF3"
```
Comment (HarelM) [inline src/reference/v8.json:70]: Can we make the example a bit more descriptive, i.e. add more parameters?

File: src/validate/validate_sky.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import ValidationError from '../error/validation_error';
+import getType from '../util/get_type';
+import validate from './validate';
+
+export default function validateSky(options) {
```
Comment (HarelM) [inline src/validate/validate_sky.ts:5]: Can you please add a test to cover this validation code?

**Comment by HarelM** [comment]
I'm merging this into a local branch, I'll continue to work on it and create a PR that addresses my review comments.

**Pull request merged**